siege_utilities/geo/providers/nces_download.py

```
@@ -0,0 +1,388 @@
+            # Normalize column names
+            col_upper = {c: c.upper() for c in df.columns}
+            df = df.rename(columns=col_upper)
+
+            lat_col = next((c for c in ("LAT", "LATITUDE", "LAT1920") if c in df.columns), None)
+            lon_col = next((c for c in ("LON", "LONGITUDE", "LON1920") if c in df.columns), None)
+
+            if lat_col is None or lon_col is None:
+                raise NCESDownloadError(
+                    f"Cannot find lat/lon columns in {csv_path}. "
+                    f"Available: {list(df.columns)}"
+                )
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)